src/main/java/net/sf/jabref/importer/fileformat/MedlineImporter.java

```diff
@@ -84,50 +140,532 @@ public boolean isRecognizedFormat(BufferedReader reader) throws IOException {
+                fields.put("sections", join(result, "; "));
+            }
+            if (bookDocument.getKeywordList() != null) {
+                addKeyWords(fields, bookDocument.getKeywordList());
+            }
+            if (bookDocument.getContributionDate() != null) {
+                addContributionDate(fields, bookDocument.getContributionDate());
+            }
+            if (bookDocument.getPublicationType() != null) {
+                List<String> result = new ArrayList<>();
+                for (PublicationType type : bookDocument.getPublicationType()) {
+                    if (type.getContent() != null) {
+                        result.add(type.getContent());
+                    }
+                }
+                fields.put("pubtype", join(result, ", "));
+            }
+            if (bookDocument.getArticleTitle() != null) {
+                ArticleTitle articleTitle = bookDocument.getArticleTitle();
+                ArrayList<String> titles = new ArrayList<>();
+                for (Serializable content : articleTitle.getContent()) {
+                    if (content instanceof String) {
+                        titles.add((String) content);
+                    }
+                }
+                fields.put("article", join(titles, ", "));
+            }
+            if (bookDocument.getBook() != null) {
+                addBookInformation(fields, bookDocument.getBook());
+            }
+        }
+
+        if (currentArticle.getPubmedBookData() != null) {
+            PubmedBookData bookData = currentArticle.getPubmedBookData();
+            putIfNotNull(fields, "pubstatus", bookData.getPublicationStatus());
+        }
+
+        BibEntry entry = new BibEntry(IdGenerator.next(), "article"); // id assumes an existing database so don't create one here
+        entry.setField(fields);
+
+        bibItems.add(entry);
+    }
+
+    private void addBookInformation(HashMap<String, String> fields, Book book) {
+        if (book.getPublisher() != null) {
+            Publisher publisher = book.getPublisher();
+            putIfNotNull(fields, "publocation", publisher.getPublisherLocation());
+            putStringFromSerializableList(fields, "publisher", publisher.getPublisherName().getContent());
+        }
+        if (book.getBookTitle() != null) {
+            BookTitle title = book.getBookTitle();
+            putStringFromSerializableList(fields, "title", title.getContent());
+        }
+        if (book.getPubDate() != null) {
+            addPubDate(fields, book.getPubDate());
+        }
+        if (book.getAuthorList() != null) {
+            List<AuthorList> authorLists = book.getAuthorList();
+            //authorLists size should be one
+            if (authorLists.size() == 1) {
+                for (AuthorList authorList : authorLists) {
+                    handleAuthors(fields, authorList);
+                }
+            }
+        }
+
+        putIfNotNull(fields, "volume", book.getVolume());
```

Comment:
In #1579 we discussed this, and we aim to create a class with all the field names as constants in the model.